staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go

```diff
@@ -349,3 +350,12 @@ type StorageVersionProvider interface {
 	// list of kinds the object might belong to.
 	StorageVersion() runtime.GroupVersioner
 }
+
+// ResetFields maps versions to the sets of fields that will be reset by the storage strategy.
+type ResetFields map[fieldpath.APIVersion]*fieldpath.Set
```

Comment:
I find this a little confusing, and I suspect it could lead to more confusion later.

`rest` is pulling SMD, and then `fieldmanager` is pulling `rest`, so you end-up having `rest` squeezed in the middle, that seems a little wrong. Can we have this type defined in `fieldmanager` (or don't we have a smd equivalent?)?

Reply:
I agree, I didn't add a type in SMD I think, but it makes sense to have one in the fieldManager.
The reason I put it into rest is, that rest is already imported in most places anyways.

Reply:
And that the type defines the fields reset by the rest strategy

Reply:
> And that the type defines the fields reset by the rest strategy

It's fine since the requirement is coming from somewhere else.

Squeezing `rest` between `fieldmanager` and `smd` would be unfortunate, especially just for a type definition.

Reply:
Ideally, this would have come from SMD (https://github.com/kubernetes-sigs/structured-merge-diff/blob/master/merge/update.go#L34)

Reply:
Or just don't define a type :-)